### PR TITLE
support python 3

### PR DESCRIPTION
**Pull request by rzimmerman**
This includes some minor cleanup, `six` machinations, and package updates to get this to work in Python 3.

It passes unit tests in cPython 2.7.11 and 3.5.1.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,10 +2,11 @@ language: python
 
 python:
 - '2.7'
+- '3.5'
 
 env:
   - EXTRAS='[test]'
-  - EXTRAS='[test,s3,test-s3]'
+  - EXTRAS='[test,s3,test_s3]'
```
Comment (bcavagnolo) [inline .travis.yml]: Hey @rzimmerman can you advise on why this change is necessary? And why a corresponding change is apparently not needed in setup.py?

Comment (rzimmerman) [inline .travis.yml]: I'm honestly not sure. For some reason running pip install with the dash in python 3 produces:

```(dlcommon3) ~/datalake-common (py3 ✔) ᐅ pip install -e ".[test,s3,test-s3]"
Obtaining file:///Users/rob/datalake-common
  datalake-common 0.20+6.ge4c7b25 does not provide the extra 'test-s3'```

Whereas with the underscore it's fine.

Something must be translating whatever's in the setup.py. Weirdly I don't see it happen in python 2.

I can patch this to change the setup.py, too. I'm not sure what's translating the dash but this should at least be consistent.
